tests/unit/defrag.cpp

```diff
@@ -0,0 +1,396 @@
+
+	iterator_private &ipriv = *static_cast<iterator_private*>(priv);
+	auto &items = ipriv.items;
+	BOOST_REQUIRE(ipriv.number_checked < items.size()); // index out of range
+
+	auto &item = items[ipriv.number_checked];
+	BOOST_REQUIRE(!item.removed);  // item removed
+	BOOST_REQUIRE(!item.checked); //  item already checked
+	BOOST_REQUIRE(dc->data_size == item.value.size());  // sizes mismatch
+
+	std::vector<char> data(dc->data_size);
+	int ret = __eblob_read_ll(fd, data.data(), dc->data_size, data_offset);
+	BOOST_REQUIRE(ret == 0);  // can't read data
+	BOOST_REQUIRE(data == item.value);  // contet of value differ
+
+	item.checked = true;
+	++ipriv.number_checked;
+	return 0;
+}
+
+
+int datasort(eblob_wrapper &wrapper, const std::set<size_t> &indexes) {
+	size_t number_bases = indexes.size();
+	assert(!indexes.empty());
+	assert(*indices.rbegin() < number_bases);
+
+	datasort_cfg dcfg;
+	memset(&dcfg, 0, sizeof(datasort_cfg));
+	std::vector<eblob_base_ctl *> bctls(number_bases);
+	eblob_base_ctl *bctl;
+	size_t index = 0;
+	size_t loop_index = 0;
+	list_for_each_entry(bctl, &wrapper.get()->bases, base_entry) {
+		if (indexes.count(loop_index)) {
+			bctls[index] = bctl;
+			++index;
+		}
+
+		++loop_index;
+		if (index == number_bases)
+			break;
+	}
+
+	init_datasort_cfg(wrapper, dcfg);
+	dcfg.bctl = bctls.data();
+	dcfg.bctl_cnt = bctls.size();
+	return eblob_generate_sorted_data(&dcfg);
+}
+
+
+int iterate(eblob_wrapper &wrapper,
+	    iterator_private &priv) {
+	eblob_index_block full_range;
+	memset(full_range.start_key.id, 0, EBLOB_ID_SIZE);
+	memset(full_range.end_key.id, 0xff, EBLOB_ID_SIZE);
+
+	eblob_iterate_callbacks callbacks;
+	memset(&callbacks, 0, sizeof(callbacks));
+	callbacks.iterator = iterate_callback;
+
+	eblob_iterate_control ictl;
+	memset(&ictl, 0, sizeof(struct eblob_iterate_control));
+	ictl.b = wrapper.get();
+	ictl.log = ictl.b->cfg.log;
+	ictl.flags = EBLOB_ITERATE_FLAGS_ALL | EBLOB_ITERATE_FLAGS_READONLY;
+	ictl.iterator_cb = callbacks;
+	ictl.priv = &priv;
+	ictl.range = &full_range;
+	ictl.range_num = 1;
+	return eblob_iterate(wrapper.get(), &ictl);
+}
+
+
+/**
+ * 1) Make two bases with 100 records each.
+ *    State: data.0(unsorted, 100 records), data.1(unsorted, 100 records)
+ * 2) Defrag first base
+ *    State: data.0(sorted, 100 records), data.1(unsorted, 100 records)
+ * 3) Remove half of first base.
+ *    State: data.0(sorted, 50 records), data.1(unsorted, 100 records)
+ * 4) Defrag first base.
+ *    State: data.0(sorted with view, 50 records), data.1(unsorted, 100 records)
+ * 5) Check that bases contains all 150 records
+ */
+BOOST_AUTO_TEST_CASE(first_base_sorted_second_base_unsorted) {
+	const size_t RECORDS_IN_BLOB = 100;
+	const size_t TOTAL_RECORDS = 2 * RECORDS_IN_BLOB;
+	const size_t RECORDS_TO_REMOVE = RECORDS_IN_BLOB / 2;
+
+	auto config = make_default_config();
+	initialize_eblob_config(config);
+	config.records_in_blob = RECORDS_IN_BLOB;
+	eblob_wrapper wrapper(config);
```

Comment:
Let's make local static function for creating config for defrag tests and use it:
```suggestion
	eblob_wrapper wrapper(make_config(RECORDS_IN_BLOB));
```